lib/datastore.js

```diff
@@ -420,6 +420,50 @@ class DataStore {
 
     return item;
   }
+  /**
+   * Touches an existing item in this DataStore.
+   *
+   * `{item}` is expected to be a complete object. API users should call
+   * {@link #get}, then pass the returned value to `{touch}`
+   *
+   * @param {Object} item The item to touch
+   * @returns {Object} The updated item
+   * @throws {Error} if this item does not exist
+   * @throws {TypeError} if `item` is not an object with a `id` member
+   * @throws {DataStoreError} if the `item` violates the schema
+   */
+  async touch(item) {
+    checkState(this);
+
+    let self = instance.get(this);
+    if (!item || !item.id) {
+      throw new DataStoreError(DataStoreError.INVALID_ITEM);
+    }
+
+    let id = item.id;
+    let record = await self.ldb.items.get(id);
+    let orig, encrypted;
+    if (!record) {
+      throw new DataStoreError(DataStoreError.MISSING_ITEM);
+    } else {
+      encrypted = record.encrypted;
+    }
+
+    orig = await self.keystore.unprotect(id, encrypted);
+
+    orig.last_used = new Date().toISOString();
+    encrypted = await self.keystore.protect(orig);
+
+    record = {
+      id,
+      encrypted,
+      last_used: record.last_used
+    };
+
+    await self.ldb.items.put(record);
+    self.recordMetric("touched", item.id);
```

Comment:
this might require an update to the extension's metrics reporting ...

Reply:
this comment is really a note to go check the extension and ensure it won't cause additional failures.

Reply:
>this might require an update to the extension's metrics reporting
>this comment is really a note to go check the extension and ensure it won't cause additional failures

@linuxwolf Good call. @sashei is going to do a quick smoke test before merging and have a follow-up here to do corresponding follow-on work: https://github.com/mozilla-lockbox/lockbox-extension/issues/661